Fix `yml` package release actions

# Package release actions finish with errors

## Describe the bug

The automated package publishing actions don't work well.

## To Reproduce

Steps to reproduce the behavior:

1. NPM publish [error](https://github.com/ViorelMocanu/astro-simpleanalytics-plugin/actions/runs/6645728160/job/18057687520#step:9:9)
2. GitHub publish [error](https://github.com/ViorelMocanu/astro-simpleanalytics-plugin/actions/runs/6645728157/job/18057687537#step:9:9)

```
ERR_PNPM_GIT_UNKNOWN_BRANCH  The Git HEAD may not attached to any branch, but your "publish-branch" is set to "master|main".
```

## Potential fix

[This article](https://github.com/pnpm/pnpm/issues/5894#issuecomment-1519644041) suggests at least the [NPM package publish](https://github.com/ViorelMocanu/astro-simpleanalytics-plugin/blob/main/.github/workflows/npm-publish.yml) would be fixed by changing the key name from `NODE_AUTH_TOKEN` to `NPM_TOKEN`.

Alternative is to run these as the error message says:

> If you want to disable Git checks on publish, set the "git-checks" setting to "false", or run again with "--no-git-checks".

Fixing this will allow us to publish the packages automatically both on GitHub and on NPM.

--- a/.github/workflows/npm-publish-github-packages.yml
+++ b/.github/workflows/npm-publish-github-packages.yml
@@ -51,6 +51,6 @@ jobs:
       - name: Test
         run: pnpm test
       - name: Publish
-        run: pnpm publish
+        run: pnpm publish --no-git-checks
         env:
           NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}

--- a/.github/workflows/npm-publish.yml
+++ b/.github/workflows/npm-publish.yml
@@ -56,3 +56,4 @@ jobs:
         run: pnpm publish --provenance --access public
         env:
           NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
+          NPM_TOKEN: ${{secrets.NPM_TOKEN}}

